skyportal/handlers/api/spectrum.py

```diff
@@ -790,3 +797,90 @@ def get(self):
 
         self.verify_and_commit()
         return self.success(data=query.all())
+
+
+class SyntheticPhotometryHandler(BaseHandler):
+    @auth_or_token
+    def post(self, spectrum_id):
+        """
+        ---
+        description: Create synthetic photometry from a spectrum
+        tags:
+          - spectra
+        parameters:
+          - in: path
+            name: spectrum_id
+            required: true
+            schema:
+              type: integer
+          - in: query
+            name: filters
+            schema:
+              type: list
+            required: true
+            description: |
+                List of filters
+        responses:
+          200:
+            content:
+              application/json:
+                schema: SingleSpectrum
+          400:
+            content:
+              application/json:
+                schema: Error
+        """
+
+        data = self.get_json()
+        filters = data.get('filters')
+
+        spectrum = Spectrum.get_if_accessible_by(
+            spectrum_id,
+            self.current_user,
+            raise_if_none=True,
+        )
+
+        spec_dict = recursive_to_dict(spectrum)
+        wav = spec_dict['wavelengths']
+        flux = spec_dict['fluxes']
+        err = spec_dict['errors']
+        print(spectrum.astropy_units)
+        obstime = spec_dict['observed_at']
+
+        try:
+            spec = sncosmo.Spectrum(
+                wav, flux * spectrum.astropy_units, err * spectrum.astropy_units
+            )
+        except TypeError:
+            spec = sncosmo.Spectrum(wav, flux * spectrum.astropy_units)
+
+        data_out = []
+        for filt in filters:
+            try:
+                mag = spec.bandmag(filt, magsys='ab')
+                magerr = 0
+            except ValueError:
+                continue
+
+            data_out.append(
+                {
+                    'mjd': Time(obstime, format='datetime').mjd,
+                    'ra': spectrum.obj.ra,
+                    'dec': spectrum.obj.dec,
+                    'mag': mag,
+                    'magerr': magerr,
+                    'filter': filt,
+                    'limiting_mag': 25.0,
+                }
+            )
+        df = pd.DataFrame.from_dict(data_out)
+        df['magsys'] = 'ab'
+        data_out = {
```

Comment:
I'd rename one of the `data_out` variables.

Reply:
yup.